pkgs/applications/editors/neovim/default.nix

```
@@ -16,7 +21,10 @@ let
     ++ optionals doCheck [
         nvim-client luv coxpcall busted luafilesystem penlight inspect
       ]
+    ++ optional devMode luacheck
```

So here, I'd suggest:

```suggestion
    ++ optional doCheck luacheck
```

---

luacheck is just for linting, it's not used during the functionaltest (it's deceptive :) )